warp/_src/context.py

```
@@ -2799,6 +2799,20 @@ def add_ref(ref):
+                # A function bound to a kernel-local (e.g. `f = mod.func`) is later called
+                # through the local, so the ast.Call above resolves to the local rather than
+                # to `mod.func`. Resolve the right-hand side here so the dependency on the
+                # bound function's module is still registered, mirroring the hashing logic in
+                # Adjoint.get_references. Without this, reloading the bound function's module
+                # would not unload the dependent kernel and a stale executable could be reused.
+                try:
+                    rhs_func, _ = adj.resolve_static_expression(node.value, eval_types=False)
+                    if isinstance(rhs_func, warp._src.context.Function) and rhs_func.module is not None:
+                        add_ref(rhs_func.module)
+                except Exception:
+                    pass
```

Ruff (0.15.14)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)